src/Compiler/CodeGen/EraseUnions.fs

```diff
@@ -214,6 +214,18 @@ let mkUnionCaseFieldId (fdef: IlxUnionCaseField) =
     // Use the lower case name of a field or constructor as the field/parameter name if it differs from the uppercase name
     fdef.LowerName, fdef.Type
 
+let inline getFieldsNullability (g: TcGlobals) (ilf: ILFieldDef) =
+    if g.checkNullness then
+        ilf.CustomAttrs.AsArray()
+        |> Array.tryFind (IsILAttrib g.attrib_NullableAttribute)
+    else
+        None
+
+let mkUnionCaseFieldIdAndAttrs g fdef =
```

Comment:
Also worth inlining? Since in this case option allocations might be lifted to stack.

Reply:
The attribute is only present if the field is marked as nullable, otherwise it is not there -> None -> null -> empty list   == all without allocations. Assumption is most DU fields will remain non-nullable.

It is called from functions which are really big already, therefore preferred not to enforce inlining.